src/components/schedule-page/table/schedule/schedule-parts/base-cell/base-cell.models.ts

```diff
@@ -0,0 +1,42 @@
+import { VerboseDate } from "../../../../../../common-models/month-info.model";
+import { CellColorSet } from "../../../../../../helpers/colors/cell-color-set.model";
+import { BaseCellInputOptions } from "./base-cell-input.component";
+
+export type CellType = "cell" | "highlighted-cell";
+export const baseCellDataCy = (index: number, cellType: CellType): string => `${index}-${cellType}`;
```

Comment:
Dodałem tę funkcję, żeby uniknąć długich chainów w stylu

```
  cy.get(`[data-cy=${section}] data-cy=[]`)
          .children()
          .children()
          .eq(workerIdx)
          .children()
          .eq(Number(key))
          .should("contain", hoursInfo[key]);
```
Niestey, nie mogłem zamiast dawać  `data-cy` zależne od indeksów, użyć `nth-of-type`, ponieważ `nth-of-type` działa dla dzieci w obrębie jednego rodzica, a w naszym przypadku tak nie jest.: u nas każda komórka jest dzieckiem conajmniej jednego `td`, przez co np.  `nth-of-type(10)` żeby wybrać 10 komórkę w wierszu, nie zadziałało by.



Chociaż jak teraz myślę, to pewnie można by było wynieść `data-cy`,  na poziom tego `td` o którym napisałem wyżej, ale szczerze mówiąc nie chcę próbować to robić, ponieważ obecna implementacja niczym nie szkodzi i nie chciałbym też teraz tracić czas na to, żeby to przepisywać.

Ale jakbyś uważał, że ma sens wprowadzić taką zmianę, to mogę nad tym siąść.

Reply:
E nie i tak jest już lepiej